frontend/src/app/user/login/login.component.ts

```diff
@@ -1,82 +1,73 @@
-import { HttpErrorResponse } from '@angular/common/http';
-import { Component } from '@angular/core';
-import { FormControl, Validators } from '@angular/forms';
-import { ActivatedRoute, Router } from '@angular/router';
+import { Component, DestroyRef, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
 import { AuthService } from '@services/auth.service';
-import { passwordValidators, usernameValidators } from '../validation';
-
+import { UserLogin } from '../models/user';
+import { map, merge, startWith } from 'rxjs';
+import { controlErrorMessages$, formErrorMessages$, setErrors, updateFormValidity } from '../utils';
+import { ToastService } from '@services/toast.service';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Router } from '@angular/router';
 
+type LoginForm = {
+    [key in keyof UserLogin]: FormControl<string>;
+}
 
 @Component({
   selector: 'lc-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss']
 })
-export class LoginComponent {
-    usernameInput = new FormControl('', [
-        Validators.required,
-        ...usernameValidators
-    ]);
-    usernameErrorMessage = '';
-    passwordInput = new FormControl('', [
-        Validators.required,
-        ...passwordValidators,
-    ]);
-    passwordErrorMessage = '';
+export class LoginComponent implements OnInit {
+    public form = new FormGroup<LoginForm>({
+        username: new FormControl<string>('baltus', {
+            nonNullable: true,
+            validators: [Validators.required]
+        }),
+        password: new FormControl<string>('knerbonius', {
+            nonNullable: true,
+            validators: [Validators.required]
+        }),
+    });
```

Comment:
Are `baltus` / `knerbonius` testing values you forgot to remove? The form now autofills these values, which is very weird.

Reply:
Yes, you caught this PR while it was still in draft mode that was technically not review-ready. I've obviously removed this values in a later commit (that had not been pushed yet).

(Still, they should have never made it into a commit anyway; thanks for spotting this!)